src/Ev.ServiceBus.Abstractions/Factories/IClientFactory.cs

```diff
@@ -3,8 +3,8 @@
 // ReSharper disable once CheckNamespace
 namespace Ev.ServiceBus.Abstractions
 {
-    public interface ITopicClientFactory
+    public interface IClientFactory
```

Comment:
Maybe make this generic?
```suggestion
public interface IClientFactory<TClientOptions> where TClientOptions : ClientOptions
{
        IClientEntity Create(TClientOptions options, ConnectionSettings connectionSettings);
}
```
It will remove the casting that is done in the factories.

Reply:
IClientEntity  could also be added actually as a generic.

Reply:
Done